app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,44 @@
+class PostsController < ApplicationController
+  # this is a filter to restrict access only to logged in users:
+  # NOTE that we apply ONLY so that this check/filter only occurs on the edit and :edit and :update actions
+  before_action :logged_in_user, only: [:new, :create]
+
+  def new
+    @post = Post.new
+  end
+
+  def show
+    @post = Post.find(params[:id])
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+
+    if @post.save
+      flash[:success] = 'Your gossip has been posted'
+      redirect_to root_url
+      # root_url will just be where all the posts are shown
+    else
+      render :new
+    end
+  end
+
+  # to show all the posts:
```

Comment:
In general, code should be clear enough to be understood without additional comments ;)

So, if you leave some comment you should double check if:

- code could be written in better, more clear way
- comment is redundant 

In this case, I think the latter;)

----
P.S. sometimes code is super clear but comment still seems to be reasonable: like here: Do not leave comments in code, unless they really explains sth like here: https://github.com/PolarisTLX/members-only/pull/1/files#diff-ab4e9273f7df75aa5004a42f757d6f51R20

Life is hard! :trollface: